### ISSUE TITLE
-from should support "last squash" or "last from" so it's easier to build multiple squashed layers on top of each other.

### ISSUE THREAD

**Issue by cetex** [jwilder/docker-squash#20]
Hi!

I'm building an environment where I have a minimal base-image, i then add layers on top of it to keep the required image size small so we can keep the filesystem usage on the servers (which is in ram) as low as possible.

When i squash the build of base-image/biglayer1/biglayer2/biglayer3 where i only want to squash biglayer3 docker-squash will squash it down to a new layer directly on top of biglayer1, so now i have biglayer2, and biglayer2&3 in two images, not biglayer2 and the addon biglayer3 where needed.

This is what it looks like when only first and second layer (java, java/mesos, java/zookeeper) has been squashed. every layer depends on the previous one.

```
docker images -a --tree
Warning: '--tree' is deprecated, it will be removed soon. See usage.
└─16a45869d60a Virtual Size: 95.27 MB Tags: trusty/minbase:latest
  └─0131ed4dbe05 Virtual Size: 414.3 MB Tags: trusty/java:latest
    ├─4251ca7211b9 Virtual Size: 418.9 MB
    │ └─e7f4bca2c41a Virtual Size: 418.9 MB Tags: trusty/zookeeper:latest
    └─840bad29bff6 Virtual Size: 577.8 MB
      └─4435bbaee79f Virtual Size: 577.8 MB Tags: trusty/mesos:latest
        └─82c13b47631d Virtual Size: 723.7 MB
          └─33621f55ac5b Virtual Size: 751 MB
            └─1c9e071a8c56 Virtual Size: 1.145 GB
              └─0a8936865afc Virtual Size: 1.176 GB
                └─30d31d294d37 Virtual Size: 1.176 GB
                  └─518bca946236 Virtual Size: 1.177 GB
                    └─76430b0182b9 Virtual Size: 1.178 GB
                      └─d4667f91fa20 Virtual Size: 1.178 GB
                        └─49497d6ebcd4 Virtual Size: 1.178 GB
                          └─ed12d73f94c3 Virtual Size: 1.178 GB
                            └─f16a5b30bcaa Virtual Size: 1.178 GB
                              └─e67fd997dbe8 Virtual Size: 1.178 GB Tags: trusty/aurora_tmp:latest
```

If i squash aurora it will end up directly under the java-layer, not under java/mesos/aurora where it belongs.

```
docker save trusty/aurora_tmp | sudo ./docker-squash -t trusty/aurora | docker load
docker images -a --tree
Warning: '--tree' is deprecated, it will be removed soon. See usage.
└─16a45869d60a Virtual Size: 95.27 MB Tags: trusty/minbase:latest, 127.0.0.1:5000/trusty/minbase:latest
  └─0131ed4dbe05 Virtual Size: 414.3 MB Tags: trusty/java:latest
    ├─037c9eaf296d Virtual Size: 709.4 MB
    │ └─b995defbf385 Virtual Size: 709.4 MB
    │   └─b009ced4b759 Virtual Size: 709.4 MB Tags: trusty/aurora:latest <- this should be beneath mesos-layer.
    ├─4251ca7211b9 Virtual Size: 418.9 MB
    │ └─e7f4bca2c41a Virtual Size: 418.9 MB Tags: trusty/zookeeper:latest
    └─840bad29bff6 Virtual Size: 577.8 MB
      └─4435bbaee79f Virtual Size: 577.8 MB Tags: trusty/mesos:latest
```

If the "-from" flag supported something like "last-defined from" or "first layer after last squash" it would be layering aurora on top of mesos properly instead of creating a new aurora&mesos layer.

**Comment by palamccc** [comment]
**As of now -from option accepts only id as argument**

But this is not very convenient, because if base image is updated, id changes, and we have to update id in -from option in our scripts. 

I looked into PR https://github.com/jwilder/docker-squash/pull/21 , but its not detecting last FROM layer properly, in my case, it only takes first Layer.  I tried to debug and that program is trying to find last "#(nop) ADD file" in layer commands, and in my case, i have only one "#(nop) ADD file" https://github.com/cetex/docker-squash/blob/master/export.go#L275-L298

**I think we need a way to specify squashing boundary, such that it can automated**

Following is my proposal, we can add an additional echo command in Dockerfile just to mark the squash -from layer, 

    FROM ubuntu
    RUN ....
    RUN ...
    RUN echo my-custom-squash-boundary
    RUN ....
    RUN ...
    CMD ....

and later in docker-squash command line, we can specify <code>-fromPattern my-custom-squash-boundary</code>, and docker-squash can search for layer matching the string.

**If we add feature like this, docker squash can be used in CI to squash specific layers.**
